### ISSUE TITLE
ForumPost acts_as_votable

### ISSUE THREAD

**Issue by perlmunger** [excid3/simple_discussion#22]
This may be a rails question more than simple discussion (again, sorry), but is it possible to add something like acts_as_votable to the ForumPost model from within my rails apps rather than within simple_disucssion itself?

**Comment by excid3** [comment]
Yeah, totally. Should just add the gem to your model and customize the views for it. You'll probably need to build your own links and controller action for the votes up and down, but shouldn't be too hard.

**Comment by perlmunger** [comment]
Right. I've customized views and controllers already. (BTW, thanks for this great gem. It's really intuitive and super simple to customize). I guess (lack of Rails fluency showing here) I'm not sure exactly what you mean by "Should just add the gem to your model". Are you saying I create my own model that wraps the same table as the SimpleDiscussion ForumPost model does? Or is there a way to just create an extension to that class and add the acts_as_votable to it?

**Comment by excid3** [comment]
You'll need to re-open the ForumPost class from the gem so that you can add acts_as_votable to it.

The easiest way is probably to add an initializer and define your code there. I think something like this should reopen the class safely.

```
# config/initializers_simple_discussion.rb
class ForumPost
  acts_as_votable
end
```

**Comment by perlmunger** [comment]
So I think you intended for that file path to be `# config/initializers/simple_discussion.rb`, right?

I tried it in the intializers directory and I get:

> config/initializers/simple_discussion.rb:2:in `<class:ForumPost>': undefined local variable or method `acts_as_votable' for ForumPost:Class (NameError)

when I attempt to run the rails server. This is rails 6, so I may be running into something specific to my configuration. 

(I made sure acts_as_votable is in the Gemfile and successfully bundle installed. I also successfully ran its migration)

Not sure. Any ideas?

**Comment by excid3** [comment]
Yes, and you probably have to inherit from ApplicationRecord then if you're getting that error.

```ruby
# config/initializers/simple_discussion.rb
class ForumPost < ApplicationRecord
  acts_as_votable
end
```

**Comment by epetkus** [comment]
@excid3 I know this is an old one - I am actually on the same boat as this original thread. So I wont to make `ForumThread` to have `act_as_voteable` and as per your last suggestion, I am adding a new initializer to add `act_as_voteable` but then it seems like any other gems are not available. Example error:
```
undefined method `friendly' for #<Class:0x0000000005531a10> Did you mean? friendly_id?
```

Is there a way to load all the gems before `config/initializers/simple_discussion.rb`? (I realize this is more of me being bad at rails than anything else!) Thanks!
